### PR TITLE
Add toolbar button to set font color

### PR DESCRIPTION
**Pull request by bampakoa**
Used library https://github.com/buberdds/angular-bootstrap-colorpicker.
The button does not have an icon yet. @Velzevul said maybe he could design one.

screenshot from 2015-06-28 19 14 52

### CODE REVIEW

**Comment by besuhoff** [comment]
I can't imagine the case when font color should be changed on site.

**Comment by bampakoa** [comment]
It's a WYSIWYG editor. Every editor has this capability.

**Comment by stevermeister** [comment]
@bampakoa , 
I think @besuhoff meant that we would have to hardcode inline styles for it.

Yes, it's possible. But it should not be common case.
Now I'm in process of making conceptual design for **Plugins**, that should allow you use any buttons you want without changing base minimal set.

**Comment by besuhoff** [comment]
@bampakoa Yes, every editor has this capability because WYSIWYG editors are created by programmers, not designers. But this is obviously wrong.

When you have a site you have semantic styles on it. For example, regular text is black 14px, header one is grey 24px, error messages are red, links are blue etc.

But here comes some manager with some WYSIWYG and decides that "the text will become more attractive if you paint it light brown". But is he actually the one who decides?

WYSIWYG should provide ability to apply semantic styles on site content but not just to create a piece of Verdana painted in light brown, sized 15px and blinking just because content manager decided it would look pretty.

**Comment by bampakoa** [comment]
@besuhoff I have used the ngWig in an enterprise application, not a regular site, that I have built and the content manager was quite happy when I gave him this ability. He was able to color various part of the content that was shown in a client application of a mobile device. Why should we bind the library to certain usage only?

**Comment by bampakoa** [comment]
@stevermeister would you be interested in checking the functionality of the PR or should I close it?

**Comment by stevermeister** [comment]
@bampakoa could you please wait a bit? 
I'm currently working on possibilities to add plugins.
I would like to use your functionality like plugin if you don't mind, pretty useful.

**Comment by bampakoa** [comment]
@stevermeister no problem. I was just asking because there were a lot of days of inactivity in the PR.

**Comment by stevermeister** [comment]
@bampakoa 
we now have plugin system, would you be so kind to convert you solution into plugin?

**Comment by bampakoa** [comment]
οκ @stevermeister I will update the PR. 
Nice job by the way with the plugin system :+1:

**Comment by bampakoa** [comment]
@stevermeister you are good to go. I also added an entry in **package.json** for ***grunt-contrib-cssmin*** which I suppose should be there.

File: dist/ng-wig.js [hunk 1]
```diff
@@ -4,102 +4,102 @@
...
+        return {
+            scope: {
+                content: '=ngWig'
+            },
+            restrict: 'A',
+            replace: true,
+            templateUrl: 'ng-wig/views/ng-wig.html',
+            link: function (scope, element, attrs) {
+
+                scope.editMode = false;
```
Comment (stevermeister) [inline dist/ng-wig.js]: you did not change all this part, right?
why is it here? 

may be because of extra spaces?

File: dist/ng-wig.js [hunk 2]
```diff
@@ -4,102 +4,102 @@
...
+
+            //model --> view
+            ngModelController.$render = function () {
+                $element.html(ngModelController.$viewValue || '');
+            };
+
+            //view --> model
+            function viewToModel() {
+                ngModelController.$setViewValue($element.html());
+                //to support Angular 1.2.x
```
Comment (stevermeister) [inline dist/ng-wig.js]: this part was removed. 

may be it's easier to create new clean PR? not to resolve a lot of conflicts

File: dist/ng-wig.js (hunk 1 above)
Comment (bampakoa) [inline dist/ng-wig.js]: Yes, I messed up the **dist** directory by accident and tried to revert it back.

File: dist/ng-wig.js [hunk 3]
```diff
@@ -117,63 +117,63 @@ angular.module('ngWig')
...
 
-  var buttonLibrary = {
-    list1: {title: 'Unordered List', command: 'insertunorderedlist', styleClass: 'fa-list-ul'},
-    list2: {title: 'Ordered List', command: 'insertorderedlist', styleClass: 'fa-list-ol'},
-    bold: {title: 'Bold', command: 'bold', styleClass: 'fa-bold'},
-    italic: {title: 'Italic', command: 'italic', styleClass: 'fa-italic'},
-    link: {title: 'Link', command: 'createlink', styleClass: 'fa-link'}
-  };
+    var buttonLibrary = {
+        list1: {title: 'Unordered List', command: 'insertunorderedlist', styleClass: 'fa-list-ul'},
```
Comment (stevermeister) [inline dist/ng-wig.js]: again a lot of spaces.
it's difficult to review because of them :(

**Comment by stevermeister** [comment]
@bampakoa thx for update!
could you clean it up a bit? to make review more convenient

**Comment by bampakoa** [comment]
yeap, no problem! Sorry about the mess :worried:

**Comment by bampakoa** [comment]
Moved to #80

**Pull request closed without merging**